src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-cluster.cpp

```diff
@@ -0,0 +1,210 @@
+        if (mLogic.GetFeatureFlags().Has(Feature::kErrorCounts))
+        {
+            Attributes::BeaconLostCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiBeaconLostCount, count, encoder);
+        }
+        break;
+    }
+    case BeaconRxCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts))
+        {
+            Attributes::BeaconRxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiBeaconRxCount, count, encoder);
+        }
+        break;
+    }
+    case PacketMulticastRxCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts))
+        {
+            Attributes::PacketMulticastRxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiPacketMulticastRxCount, count, encoder);
+        }
+        break;
+    }
+    case PacketMulticastTxCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts))
+        {
+            Attributes::PacketMulticastTxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiPacketMulticastTxCount, count, encoder);
+        }
+        break;
+    }
+    case PacketUnicastRxCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts))
+        {
+            Attributes::PacketUnicastRxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiPacketUnicastRxCount, count, encoder);
+        }
+        break;
+    }
+    case PacketUnicastTxCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kPacketCounts))
+        {
+            Attributes::PacketUnicastTxCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiPacketUnicastTxCount, count, encoder);
+        }
+        break;
+    }
+    case CurrentMaxRate::Id: {
+        if (mLogic.GetEnabledAttributes().enableCurrentMaxRate)
+        {
+            Attributes::CurrentMaxRate::TypeInfo::Type rate;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiCurrentMaxRate, rate, encoder);
+        }
+        break;
+    }
+    case OverrunCount::Id: {
+        if (mLogic.GetFeatureFlags().Has(Feature::kErrorCounts))
+        {
+            Attributes::OverrunCount::TypeInfo::Type count;
+            return mLogic.ReadIfSupported(&DiagnosticDataProvider::GetWiFiOverrunCount, count, encoder);
+        }
+        break;
+    }
+    default:
+        break;
+    }
+    return Protocols::InteractionModel::Status::UnsupportedAttribute;
+}
+
+CHIP_ERROR WiFiDiagnosticsServerCluster::Attributes(const ConcreteClusterPath & path,
+                                                    ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
+{
+    ReturnErrorOnFailure(builder.EnsureAppendCapacity(13 + DefaultServerCluster::GlobalAttributes().size()));
```

Comment:
use https://github.com/project-chip/connectedhomeip/blob/master/src/app/server-cluster/AttributeListBuilder.h to save some flash